processing/data_collection/gazette/spiders/rn_mossoro.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+
+import dateutil
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class RnMossoroSpider(BaseGazetteSpider):
+
+    name = "rn_mossoro"
+    allowed_domains = ["jom.prefeiturademossoro.com.br"]
+    start_urls = ["http://jom.prefeiturademossoro.com.br/todas-as-edicoes-2/?_page=1"]
+
+    TERRITORY_ID = "2408003"
+
+    def parse(self, response):
+        for entry in response.css(".pt-cv-ifield"):
+
+            url = entry.css("a:first-of-type::attr(href)").get()
+            datetime_meta = entry.css("span.entry-date time::attr(datetime)").get()
+            date = dateutil.parser.isoparse(datetime_meta).date()
+
+            yield scrapy.Request(url, meta={"date": date}, callback=self.parse_gazette)
+
+        page = int(response.css(".pt-cv-pagination::attr(data-currentpage)").get())
+        total_pages = int(
+            response.css(".pt-cv-pagination::attr(data-totalpages)").get()
+        )
+
+        if page == 1:
+            for i in range(2, total_pages + 1):
```

Comment:
Given the conversation here https://github.com/okfn-brasil/querido-diario/issues/247
We should avoid yielding all pages upfront and walk them one each time, or something like that, to avoid making unnecessary requests

Reply:
Refactored. On my tests now it issues about 3 to 4 additional requests when compared to the amount of scraped items.